api/src/opentrons/algorithms/graph.py

```
@@ -0,0 +1,190 @@
+
+    @classmethod
+    def build(cls, graph: List[VertexLike],
+              sort_by: Callable = default_sort) -> Graph:
+        """
+        Graph class builder.
+
+        :param graph: A list of nodes to add to the graph.
+        :param sort_by: The function used to sort the graph
+        in priority order.
+        :returns: A graph class
+        """
+        sorted_graph = []
+        lookup_table = {}
+        for vertex in graph:
+            vertex_obj = cls.build_vertex(vertex)
+            lookup_table[vertex.name] = vertex_obj
+            sorted_graph.append(vertex_obj)
+            sorted_graph.sort(key=sort_by)
+        return cls(sorted_graph, lookup_table, sort_by)
+
+    @property
+    def graph(self) -> Sequence[Vertex]:
+        """
+        Graph class property: graph.
+
+        :returns: A list of sorted vertex objects
+        """
+        return self._sorted_graph
+
+    @staticmethod
+    def build_vertex(vertex: VertexLike) -> Vertex:
+        """
+        Build a vertex.
+
+        Use this to sort the neighbors and then build
+        a vertex using a node dataclass.
+        :param vertex: A node dataclass
+        :returns: vertex object
+        """
+        vertex.sub_names.sort()
+        return Vertex(vertex, vertex.sub_names)
+
+    def add_vertex(self, vertex: VertexLike) -> None:
+        """
+        Add a vertex.
+
+        :param vertex: A node dataclass
+        """
+        new_vertex = self.build_vertex(vertex)
+        if new_vertex not in self._lookup_table.values():
+            self._lookup_table[new_vertex.name] = new_vertex
+            self._sorted_graph.append(new_vertex)
+            self._sorted_graph.sort(key=self._sort_by)
+
+    def remove_vertex(self, vertex: VertexLike) -> None:
+        """
+        Remove a vertex.
+
+        :param vertex: A node dataclass
+        """
+        vertex_to_remove = self._lookup_table[vertex.name]
+        del self._lookup_table[vertex.name]
```

this will cause a `KeyError` if you delete a non-existent vertex. That may be fine, but it doesn't match the behavior of `add_vertex`, which silently does nothing if you add a vertex that already exists.